src/providers/java_maven.js

```diff
@@ -77,25 +74,36 @@ export default class Java_maven extends Base_java {
 		// get custom maven path
 		let mvn = getCustomPath('mvn', opts)
 		// verify maven is accessible
-		this._invokeCommand(`${handleSpacesInPath(mvn)} --version`,'mvn is not accessible')
+		this._invokeCommand(mvn, ['--version'], error => {
+			if (error.code === 'ENOENT') {
+				throw new Error(`maven not accessible at "${mvn}"`)
+			} else {
+				throw new Error(`failed to check for maven: ${error.stderr}`)
```

Comment:
`error.stderr` would be `undefined` if there was an issue spawning the subprocess, I think for most invocations doing this should be fine (we verify if we can execute maven here and assume nothing happens that would cause issues later :upside_down_face: ) but maybe for this one we need something better?

Reply:
What do you suggest?
Let's assume worst case and cover any possible failure while running the process or spawning it

Reply:
Decided to instead attach the original error to our custom-message error, the application-side (aka not library side) should handle presenting stderr/stdout accordingly (print to extension host output/include in notification/whatever else is decided).